### PR TITLE
add shared `@apollo/experimental-nextjs-app-support` entry point

### PR DESCRIPTION
**Pull request by phryneas**
This unifies `@apollo/experimental-nextjs-app-support/rsc` and `@apollo/experimental-nextjs-app-support/ssr` into `@apollo/experimental-nextjs-app-support`.

Once we rename the package to remove the `experimental` label, this will be the only entry point.

Also, some imports are changed:
* the `useQuery`/`useSuspenseQuery` etc. hooks have been dropped. They should be imported from `@apollo/client` instead.
* `NextSSRApolloClient` has been renamed to `ApolloClient` and is available in different variations from all environments, including RSC.
* `NextSSRInMemoryCache` has been renamed to `InMemoryCache` and is available in different variations from all environments, including RSC.
* `resetNextSSRApolloSingletons` has been renamed to `resetApolloClientSingletons`

### CODE REVIEW

File: packages/experimental-nextjs-app-support/README.md [hunk 1]
```diff
@@ -105,21 +109,11 @@ function makeClient() {
...
-        ? ApolloLink.from([
-            // in a SSR environment, if you use multipart features like
-            // @defer, you need to decide how to handle these.
-            // This strips all interfaces with a `@defer` directive from your queries.
-            new SSRMultipartLink({
-              stripDefer: true,
-            }),
-            httpLink,
-          ])
-        : httpLink,
```
Comment (phryneas) [inline packages/experimental-nextjs-app-support/README.md:122]: I believe this is something that's copied blindly without any real need, so I'll omit the complexity here for now.

File: packages/experimental-nextjs-app-support/README.md [hunk 2]
```diff
@@ -167,9 +161,9 @@ This package uses some singleton instances on the Browser side - if you are writ
 For that, you can use the `resetNextSSRApolloSingletons` helper:
```
Comment (jerelmiller) [inline packages/experimental-nextjs-app-support/README.md:161]:
```suggestion
For that, you can use the `resetApolloClientSingletons ` helper:
```

File: packages/experimental-nextjs-app-support/src/ssr/index.ts [hunk 3]
```diff
@@ -1,35 +1,17 @@
-export { ApolloNextAppProvider } from "../ApolloNextAppProvider.js";
-export { resetManualSSRApolloSingletons as resetNextSSRApolloSingletons } from "@apollo/client-react-streaming/manual-transport";
-import { ApolloClient } from "@apollo/client-react-streaming";
-import { bundle } from "../bundleInfo.js";
 export {
   InMemoryCache as NextSSRInMemoryCache,
+  ApolloClient as NextSSRApolloClient,
```
Comment (jerelmiller) [inline packages/experimental-nextjs-app-support/src/ssr/index.ts:3]: Do we need a way to mark these as deprecated? Or is there another way we can hint that these exports will be removed in a future version?

**Review by jerelmiller — APPROVED** [review]
This looks great to me! I absolutely love the simplification here.

File: packages/experimental-nextjs-app-support/src/ssr/index.ts (hunk 3 above)
Comment (phryneas) [reply packages/experimental-nextjs-app-support/src/ssr/index.ts:3]: I was thinking about that, too.... I think I'll merge this as-is and add another PR for that.

**Pull request merged**